### ISSUE TITLE
vsphere_guest vm_disk 'type' parameter is silently ignored

### ISSUE THREAD

**Issue by gsatliawol** [ansible/ansible#18926]
_From @aioue on May 20, 2015 15:16_

##### Issue Type:

Bug Report
##### Ansible Version:

1.9.1
##### Ansible Configuration:

n/a
##### Environment:

OS X 10.10.3, vCenter 6
##### Summary:

Input to vsphere_guest module, vm_disk 'type' parameter is ignored. Does not throw errors for bad input.
##### Steps To Reproduce:

Run module with this input:

```
- name: Create other-app VM for testing
  connection: local
  vsphere_guest:
    vcenter_hostname: 110.120.113.333
    username: foo
    password: bar
    guest: other-app-{{ 9 |random}}{{ 9 |random}}{{ 9 |random}}{{ 9 |random}}{{ 9 |random}}
    vm_disk:
      disk1:
        size_gb: 32
        type: funkyChicken
        datastore: foo_VMFS_1
    vm_hardware:
      memory_mb: 8192
      num_cpus: 2
      osid: ubuntu64Guest
      scsi: lsi
    vm_nic:
      nic1:
        type: vmxnet3
        network: foo-other-app
        network_type: dvs
    vm_extra_config:
      notes: this is a test VM
    esxi:
      datacenter: foo-DC
      hostname: foo-dc2-esxi2
```
##### Expected Results:

Error saying 'funkyChicken' is not a valid disk type.
##### Actual Results:

```
<localhost> REMOTE_MODULE vsphere_guest vcenter_hostname=110.120.113.333 guest=other-app-22732 password=VALUE_HIDDEN username=foo
<localhost> EXEC ['/bin/sh', '-c', 'mkdir -p $HOME/.ansible/tmp/ansible-tmp-1432134778.87-76768885543882 && chmod a+rx $HOME/.ansible/tmp/ansible-tmp-1432134778.87-76768885543882 && echo $HOME/.ansible/tmp/ansible-tmp-1432134778.87-76768885543882']
<localhost> PUT /var/folders/s0/jy7xj1915_bg41nmvc3cjcdw0000gn/T/tmpMEX1JT TO /Users/tpai/.ansible/tmp/ansible-tmp-1432134778.87-76768885543882/vsphere_guest
<localhost> EXEC ['/bin/sh', '-c', u'LANG=en_US.UTF-8 LC_CTYPE=en_US.UTF-8 /usr/bin/python /Users/tpai/.ansible/tmp/ansible-tmp-1432134778.87-76768885543882/vsphere_guest; rm -rf /Users/tpai/.ansible/tmp/ansible-tmp-1432134778.87-76768885543882/ >/dev/null 2>&1']
changed: [localhost] => {"ansible_facts": {"hw_eth0": {"addresstype": "assigned", "ipaddresses": null, "label": "Network adapter 1", "macaddress": "00:50:56:b6:36:91", "macaddress_dash": "00-50-56-b6-36-91", "summary": "DVSwitch: 33 aa 36 50 12 b3 ba a6-1f b5 1d 2b e8 04 c2 01"}, "hw_guest_full_name": "Ubuntu Linux (64-bit)", "hw_guest_id": "ubuntu64Guest", "hw_memtotal_mb": 8192, "hw_name": "other-app-22732", "hw_processor_count": 2, "hw_product_uuid": "42360560-b591-a15d-f6bd-de2341927943", "module_hw": true}, "changed": true, "changes": "Created VM other-app-22732"}
```

_Copied from original issue: ansible/ansible-modules-core#1380_

**Comment by gsatliawol** [comment]
_From @ansibot on August 1, 2016 3:20_



@chrrrles, @rhoop, ping. This issue is waiting on your response.
[click here for bot help](https://github.com/jctanner/ansibullbot/blob/ISSUE_TRIAGE_CLASS/ISSUE_HELP.md)

**Comment by gsatliawol** [comment]
_From @ansibot on September 12, 2016 21:24_

@chrrrles, @rhoop, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by gsatliawol** [comment]
_From @ansibot on October 6, 2016 16:2_

@chrrrles, @rhoop, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by gsatliawol** [comment]
_From @ansibot on October 22, 2016 9:30_

@chrrrles, @rhoop, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by gsatliawol** [comment]
_From @ansibot on November 6, 2016 10:10_

@ansible, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by gsatliawol** [comment]
_From @ansibot on November 22, 2016 10:18_

@ansible, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by gsatliawol** [comment]
_From @ansibot on December 7, 2016 19:58_

This repository has been locked. All new issues and pull requests should be filed in https://github.com/ansible/ansible

Please read through the <a href="http://docs.ansible.com/ansible/dev_guide/repomerge.html#move-issues-and-prs-to-new-repo">repomerge</a> page in the dev guide. The guide contains links to tools which automatically move your issue or pull request to the ansible/ansible repo.

**Comment by ansibot** [comment]
@gsatliawol Greetings! Thanks for taking the time to open this issue. In order for the community to handle your issue effectively, we need a bit more information. 

Here are the items we could not find in your description:
- component name

Please set the description of this issue with this template:
https://raw.githubusercontent.com/ansible/ansible/devel/.github/ISSUE_TEMPLATE.md

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibotdev** [comment]
@gsatliawol You have not responded to information requests in this issue so we will assume it no longer affects you. If you are still interested in this, please create a new issue with the requested information.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)
